### PR TITLE
Create Next Actions & Allow Their Immediate Selection

### PR DESCRIPTION
**Pull request by JasonWarrenUK**
# What type of PR is this? (check all applicable)

- [X] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Styling
- [ ] Build
- [ ] Chore

## Description

- Allows creation of Next Actions within each unmet Need
- Displays them instantly and allows their selection
  > Uses a smoother logic than the similar add category feature in the toolkit

### Note for @jackcasstlesjones

Look at screenshot 2. CHOOSE A RULE. It's time.

## Screenshots

image

image

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [X] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [X] No, and this is why: It's the same component I couldn't write tests for before
- [ ] I need help with writing tests

## Delete branch after merge?

- [X] Yes
- [ ] No

### CODE REVIEW

File: src/app/toolkit/add-tool/components/AddToolTags.tsx
Comment (maxitect) [inline src/app/toolkit/add-tool/components/AddToolTags.tsx:1]: Something has changed here meaning I cannot select any existing tags when adding a tool

File: src/app/needs/next-actions/components/NextActionsSection.tsx [hunk 1]
```diff
@@ -1,38 +1,102 @@
...
+
+        setTriggerRerender(triggerRerender + 1);
+      } catch (error) {
+        console.error("Error creating Action:", error);
+      }
+      
+      setModalOpen(false);
+    }
+  }; */
+
```
Comment (maxitect) [inline src/app/needs/next-actions/components/NextActionsSection.tsx:62]:
```suggestion
```
If we don't need, delete [will not block merge if it is needed for a future use]

**Review by maxitect — CHANGES_REQUESTED** [review]
Cannot select existing tags when adding a tool anymore

File: src/app/needs/next-actions/components/NextActionsSection.tsx (hunk 1 above)
Comment (JasonWarrenUK) [reply src/app/needs/next-actions/components/NextActionsSection.tsx:62]: Good catch

**Comment by jackcasstlesjones** [comment]
> That didn't happen on my branch, my edits on that file are purely whitespace and formatting

it's not happening in main - did you change anything in the backend that could influence it?

**Review by maxitect — APPROVED** [review]
Grrrrrreat!

**Pull request merged**